src/include/txn_inline.h

```diff
@@ -396,34 +396,64 @@ __wt_txn_op_delete_apply_prepare_state(WT_SESSION_IMPL *session, WT_REF *ref, bo
  * __txn_op_delete_commit_apply_page_del_timestamp --
  *     Apply the correct start and durable timestamps to the page delete structure.
  */
-static WT_INLINE void
-__txn_op_delete_commit_apply_page_del_timestamp(WT_SESSION_IMPL *session, WT_REF *ref)
+static WT_INLINE int
+__txn_op_delete_commit_apply_page_del_timestamp(
+  WT_SESSION_IMPL *session, WT_TXN_OP *op, bool validate)
 {
+    WT_ADDR_COPY addr;
+    WT_DECL_RET;
     WT_PAGE_DELETED *page_del;
+    WT_REF *ref;
     WT_TXN *txn;
+    bool addr_found;
 
+    ref = op->u.ref;
     txn = session->txn;
     page_del = ref->page_del;
 
     WT_ASSERT(session, WT_REF_GET_STATE(ref) == WT_REF_LOCKED);
 
-    if (page_del != NULL && page_del->timestamp == WT_TS_NONE) {
-        page_del->timestamp = txn->commit_timestamp;
-        page_del->durable_timestamp = txn->durable_timestamp;
+    if (page_del != NULL) {
+        /* Validate the commit timestamp against the maximum durable timestamp on the page. */
+        if (validate) {
+            /*
+             * To compare against the commit timestamp, we must copy the address to obtain the
+             * latest start and stop durable timestamps. To prevent concurrent freeing, we need to
+             * acquire the split generation before accessing ref->addr.
```

Comment:
I still don't understand who can concurrently free the address if the fast truncate hasn't been committed?

Reply:
Reconciliation of a fast truncated page before the truncate operation gets committed can free the address.
1. fast truncate the page
2. A reader reads the page into the cache
3. Checkpoint reconciling the instantiated deleted page can perform update/restore eviction due to the uncommitted truncate operation and possibly free the `ref->addr`.

Reply:
makes sense. Can you add this scenario to the comment?

Reply:
Yes, added.